### PR TITLE
Fix kube gateway not working when namespace waypoint is applied

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
The old behavior occurred when sending a request to the kube gateway, ztunnel would redirect the traffic to the namespace waypoint.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Dual Stack
- [ ] Installation
- [x] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by howardjohn — CHANGES_REQUESTED** [review]
I think its actual valid to have k8s gateways that are intercepted. If they don't want to be, they should opt out/not opt-in of ambient redirection capture?

**Comment by howardjohn** [comment]
Maybe we by default opt out (but allow user to opt in) though, in the gateway deployment controller though not here?

**Comment by hanxiaop** [comment]
> Maybe we by default opt out (but allow user to opt in) though, in the gateway deployment controller though not here?

@howardjohn After thinking this, I think the fix was not very correct here, but the redirection may not need to be determined by users? To mention the situation clearly: when a namespace waypoint is applied in the same namespace as the k8s gateway, the k8s gateway does work with service-based routing, such as `gateway.istio-ingress/productpage`. However, IP-based routing does not work. For example, when I try to curl from the load balancer IP `10.64.30.20/productpage`, the ztunnel redirects the traffic to the gateway's namespace waypoint, which seems to have no specific handling for IP addresses.

**Comment by howardjohn** [comment]
I guess there are a few approaches:

1. Have the gateways opt-out of ambient, by default or documented to do so
2. Add LoadBalancer IP matches in the waypoint
3. Replace using IP entirely by hostnames in ztunnel, match on those in waypoint

I think for various reasons in ongoing API discussions (3) might be the long term path. 

Although I wouldn't think the request to the LoadBalancer IP would get redirected to waypoint.. maybe that is a bug

**Comment by hanxiaop** [comment]
> I guess there are a few approaches:
> 
> 1. Have the gateways opt-out of ambient, by default or documented to do so
> 2. Add LoadBalancer IP matches in the waypoint
> 3. Replace using IP entirely by hostnames in ztunnel, match on those in waypoint
> 
> I think for various reasons in ongoing API discussions (3) might be the long term path.
> 
> Although I wouldn't think the request to the LoadBalancer IP would get redirected to waypoint.. maybe that is a bug

@howardjohn I think option 3 makes sense to me. For now, I've revised to option 1 as a temporary fix, which provides a flexible option for users.

File: manifests/charts/istio-control/istio-discovery/files/kube-gateway.yaml [hunk 1]
```diff
@@ -47,9 +47,18 @@ spec:
   template:
     metadata:
       annotations:
+        {{- $ambientRedirectionAnnotation := strdict}}
+        {{- if eq (env "PILOT_ENABLE_AMBIENT_CONTROLLERS" "false") "true" }}
+            {{- if eq .Values.global.ambient.kubeGatewayRedirection true}}
```
Comment (howardjohn) [inline manifests/charts/istio-control/istio-discovery/files/kube-gateway.yaml:52]: I don't know we need a global value for this, we could just allow users to opt in explicitly by setting the `ambient.istio.io/redirection`?

```
if user has ambient.istio.io/redirection { use it } 
else { set ambient.istio.io/redirection=disabled }
```

?

Comment (linsun) [reply manifests/charts/istio-control/istio-discovery/files/kube-gateway.yaml:52]: +1 - would be nice we sort this out automatically for the user without configuration.

Comment (hanxiaop) [reply manifests/charts/istio-control/istio-discovery/files/kube-gateway.yaml:52]: Makes sense, will update it when I'm back.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**